bindings/experimental/distrdf/python/DistRDF/Ranges.py

```diff
@@ -311,92 +314,105 @@ def get_clustered_ranges(clustersinfiles, npartitions, treename, friend_info):
+
+            localstarts.append(min(clustersinsamefilelist).start)
+            localends.append(max(clustersinsamefilelist).end)
+            filelist.append(clustersinsamefilelist[0].filetuple.filename)
+
+        # The global start and end entries are retrieved as follows:
+        # - Take as start start the `start` attribute of the minimum
+        #   `ChainCluster` object in the current partition
+        # - Take as end the `end` attribute of the maximum `ChainCluster` object
+        #   in the current partition
+        # - Both values need to be shifted by the respective `offset` attribute
+        #   in the same `ChainCluster` instance
+        # - Then to make them "global with respect to the chain of files in the
+        #   current partition", we need to subtract the offset of the first
+        #   cluster in the current partition. That is equal to the amount of
+        #   entries in the chain up to the beginning of the file that cluster
+        #   belongs to. This is needed to maintain a reference of the entries of
+        #   the range with respect to the list of files that hold them. For
+        #   example, given the following files:
+        #         tree10000entries10clusters.root --> 10000 entries, 10 clusters
+        #         tree20000entries10clusters.root --> 20000 entries, 10 clusters
+        #         tree30000entries10clusters.root --> 30000 entries, 10 clusters
+        #   Building 2 ranges will lead to the following tuples::
+        #         TreeRange(globalstart=0,
+        #                 globalend=20000,
+        #                 localstarts = [0, 0],
+        #                 localends = [10000, 10000],
+        #                 filelist=['tree10000entries10clusters.root',
+        #                         'tree20000entries10clusters.root'],
+        #                 friendinfo=None)
+        #         TreeRange(globalstart=10000,
+        #                 globalend=50000,
+        #                 localstarts = [10000, 0],
+        #                 localends = [20000, 30000],
+        #                 filelist=['tree20000entries10clusters.root',
+        #                         'tree30000entries10clusters.root'],
+        #                 friendinfo=None)
+        #   The first `TreeRange` will read the first 10000 entries from the
+        #   first file, then switch to the second file and read the first 10000
+        #   entries. The second `TreeRange` will start from entry number 10000
+        #   of the second file up until the end of that file (entry number
+        #   20000), then switch to the third file and read the whole 30000
+        #   entries there.
+        firstclusterinpartition = partition[0]
+        partitionoffset = firstclusterinpartition.offset
+
+        lastclusterinpartition = partition[-1]
+
+        globalstart = firstclusterinpartition.start # + firstclusterinpartition.offset - partitionoffset
```

Comment:
is the comment a debugging leftover?

Reply:
I left it there as a justification for the fact that there is no `+- offset` for the first cluster in the partition. That's because the offset of the cluster is always subtracted with the offset of the first cluster in the partition so in this case they're equal.

Reply:
uuh can you write `# cluster offsets are relative to the partition, so the first cluster has no offset` or similar instead?